nh_food_and_fluid/models/food_fluid_review.py

```diff
@@ -176,6 +179,99 @@ def schedule_review(self, spell_activity):
             }
         )
 
+    @api.model
+    def add_reviews_to_periods(self, patient_id, periods):
+        """
+        Add review task data to the periods dictionaries passed from
+        get_periods_dictionaries on the F&F model
+        :param patient_id: ID for the patient we're looking up reviews for
+        :type patient_id: int
+        :param periods: list of periods
+        :type periods: list of dicts
+        :return: list of periods with review data added
+        """
+        for period in periods:
+            period['reviews'] = {}
+            for review_time in self.trigger_times:
+                period['reviews'][review_time] = self.get_review_data(
+                    patient_id, period.get('period_start_datetime'),
+                    review_time)
+        return periods
+
+    @api.model
+    def get_review_data(self, patient_id, period_start, trigger_time):
+        """
+        Get data for the review in a given period for a given patient
+        :param patient_id: Id of the patient we're looking up review for
+        :type patient_id: int
+        :param period_start: start of the period
+        :type period_start: Odoo datetime string
+        :param trigger_time: time the review was triggered (hour of day)
+        :type trigger_time: int
+        :return: dictionary of review score, state and user/reason
+        """
+        user_model = self.env['res.users']
+        review_creator = user_model.search(
+            [
+                ['login', '=', 'food_fluid_review_creator']
+            ]
+        )
+        period_start_datetime = datetime.strptime(period_start, dtf)
+        period_end_datetime = fields.datetime.context_timestamp(
+            self._cr, review_creator.id, period_start_datetime)\
+            .replace(hour=trigger_time)
+        if trigger_time < 7:
+            day_delta = timedelta(days=1)
+            period_end_datetime = period_end_datetime + day_delta
+        period_end = period_end_datetime.astimezone(pytz.utc).strftime(dtf)
+        period_end_start = \
+            period_end_datetime.astimezone(pytz.utc).strftime(dtf)
+        period_end_end = period_end_datetime + timedelta(minutes=1)
```

Comment:
Just in case of something strange that makes it take longer than a minute (database locking, database down, random fire), we could increase this margin of a minute to be all the way right up to a second before the next trigger point. Can use the `get_period_start_datetime` and subtract a second.